src/pool/srv_target.c

```diff
@@ -398,6 +396,8 @@ pool_free_ref(struct daos_llink *llink)
 	if (pool->sp_map != NULL)
 		pool_map_decref(pool->sp_map);
 
+	ds_iv_ns_put(pool->sp_iv_ns);
+
```

Comment:
It would make more sense to move the crt_group_secondary_destroy call together with the ds_iv_ns_put call, even though the internal references might make things work either way.

Reply:
Interestingly a secondary group descriptor is one of the memory leaks that persists in the server, although it seems to happen all the time and is not related to this or the other reference counting issue.

Reply:
ok, I will move sp_group destroy after ds_iv_ns_put().

Reply:
I wasn't suggesting you should do that, just noting that there's a related issue.   The group reference count issue has landed now, so the group will actually be freed in the destroy function so you do need to at least merge this with master to get a new test run.

Reply:
Wait, why does the "cont aggr" code accesses the IV NS but not holding a ds_pool reference?

This function  is where ds_pool/iv_ns is being freed.  It looks like the intention here is to stop cont_agg before free the ds_pool/iv_ns totally, instead of using reference count and do them individually.  or I misunderstand sth?

Reply:
> I wasn't suggesting you should do that, just noting that there's a related issue. The group reference count issue has landed now, so the group will actually be freed in the destroy function so you do need to at least merge this with master to get a new test run.

oh, sure, I will merge with the current master and retry, though moving cart group destroy after iv ns destroy seems correct to me.

Reply:
> This function is where ds_pool/iv_ns is being freed. It looks like the intention here is to stop cont_agg before free the ds_pool/iv_ns totally, instead of using reference count and do them individually. or I misunderstand sth?

Agreed. My question was, if the aggregation ULTs may use ds_pool and the IV NS, they should have taken ds_pool references. If they did, how could we reach pool_free_ref?

Reply:
pool_free_ref() will wait for aggregation ult to be stopped, then free the ds_pool/iv_ns, so we should be fine here. or I may miss the question?

Reply:
Yes, it seems fine. I'm just puzzled by the recent patches. We add reference counting to IV NS, which I thought was unnecessary, but we allow agg ULTs to not hold ds_pool references and wait for them in pool_free_ref instead...

Reply:
hmm, yes, agg ult stop probably should be moved out of pool_free_ref, after we have the IV refcount.